routers/user/auth.go

```diff
@@ -120,6 +127,11 @@ func SignInPost(ctx *context.Context, form auth.SignInForm) {
 	// Clear whatever CSRF has right now, force to generate a new one
 	ctx.SetCookie(setting.CSRFCookieName, "", -1, setting.AppSubUrl)
 
+	if redirectTo := ctx.Query("redirect_to"); len(redirectTo) > 0 {
```

Comment:
Have you actually tried removing it and doing the sign-in from a private
repository with 404 response ?


On Sat, Aug 27, 2016 at 01:57:22PM -0700, 无闻 wrote:
> > @@ -120,6 +127,11 @@ func SignInPost(ctx *context.Context, form auth.SignInForm) {
> >  	// Clear whatever CSRF has right now, force to generate a new one
> >  	ctx.SetCookie(setting.CSRFCookieName, "", -1, setting.AppSubUrl)
> >  
> > +	if redirectTo := ctx.Query("redirect_to"); len(redirectTo) > 0 {
> 
> Post request does not carry `redirect_to` URL query in any rate, so it is not useful here.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/gogits/gogs/pull/3448/files/01cfe12dd5ed2e16cfc6b0155759bc42cd15ca11#r76522277

-- 

  ()   Free GIS & Flash consultant/developer
  /\   https://strk.kbt.io/services.html